### PR TITLE
Add tree visualization for scHPL classifiers

### PR DESCRIPTION
**Pull request by nictru**
Follows the [tutorial for custom javascript content](https://shiny.posit.co/py/docs/custom-component-one-off.html). Inspired by [this example](https://vasturiano.github.io/3d-force-graph/example/tree/).

### CODE REVIEW

File: tree.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    return ui.div(
+        ui.HTML("<script src='//unpkg.com/3d-force-graph'></script>"),
+        tree_dep,
+        id=module.resolve_id(id),
+        class_="shiny-tree-output"
+    )
+
+class render_tree(Renderer[dict]):
+    def auto_output_ui(self):
+        return ui.ouptut_tree(self.ouptut_name)
```
Comment (serareif) [inline tree.py]: spelling mistakes: ouptut_trees and ouptput_name

File: tree.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    return output_tree("tree")
+
+@module.server
+def tree_server(input, output, session, _tree: reactive.Value[scHPL.TreeNode]):
+
+    @output
+    @render_tree
+    def tree():
+        cur_tree = _tree.get()
+        return format(cur_tree[0]) if cur_tree else {}
```
Comment (serareif) [inline tree.py:64]: can you use the tree instance as lists?

**Review by serareif — CHANGES_REQUESTED** [review]
I think a very nice implementation :)

File: tree.py (hunk 1 above)
Comment (nictru) [reply tree.py]: Oopsie

File: tree.py (hunk 2 above)
Comment (nictru) [reply tree.py:64]: The tree instance is always a list of nodes with length 1 (the root node), that's why this works this way.

Comment (serareif) [reply tree.py:64]: AAh makes sense

**Pull request merged**